### PR TITLE
Bump: support manual version

### LINKED ISSUE #435: Manually bump entire version, not just manually specify which part to increment
The [docs](https://commitizen-tools.github.io/commitizen/bump/#about) state that the version can be manually bumped but I believe this is incorrect as there's no other mention of this in the docs or in the CLI help beyond `--increment {MAJOR,MINOR,PATCH}` which is merely specifying the part to auto-increment.

Why do I want to manually specify the version to bump to? Well, I don't but unfortunately, dev versions are not supported - but I still want to make use of commitizen's version-updating inside of files. Otherwise, I will need to use a separate tool in conjunction, such as [bumpversion](https://github.com/c4urself/bump2version).

I want to be able to do `cz bump 1.0.1` for example (optional positional arg or `--manual <foo>` option would be fine).

A question on this though - whilst commitizen does not support dev versions, will it still be able to successfully parse a PEP440-compatible version that includes a dev version, for the sake of bumping? I.e. will commitizen be able to bump `1.0.1.dev0+foo.bar.baz` to `1.0.2` or `1.1.0` etc.?

And the ultimate related question - is there any planned support for dev versions?

### PR DESCRIPTION
**Pull request by jakob-keller**
## Description

Closes #435 

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Expected behavior

`cz bump 1.2.3` uses 1.2.3 as new version

### CODE REVIEW

**Comment by jakob-keller** [comment]
Not sure, why these tests fail. Anything I need to fix?

File: commitizen/cli.py [hunk 1]
```diff
@@ -183,6 +183,13 @@
                         "default": False,
                         "help": "retry commit if it fails the 1st time",
                     },
+                    {
+                        "name": "manual_version",
```
Comment (Lee-W) [inline commitizen/cli.py:187]: Should we make it `--manual-version`? 🤔

Comment (jakob-keller) [reply commitizen/cli.py:187]: I would not mind, but believe that makes no practical difference: 

1. decli/argparse normalises the argument name, i.e. bump.py uses `manual_version = self.arguments["manual_version"]` either way.
2. `cz bump --help` does not display the argument name, but the metavar value, i.e. `MANUAL_VERSION`.

Let me know, if you want me to change the PR.

Comment (woile) [reply commitizen/cli.py:187]: I prefer it like the proposal from this PR, as it is inline with the way changelog works: `cz changelog 1.2.3` would generate the changelog for that version. I would expect `cz bump 1.2.3` to bump the given version as well.

Comment (Lee-W) [reply commitizen/cli.py:187]: If that's the case, let's go with it 👍

**Review by Lee-W — APPROVED** [review]

**Comment by woile** [comment]
Tests are not failing, it is the coverage report. But I'm not sure why it's decreasing, it doesn't look like it's decreasing

**Comment by Lee-W** [comment]
> Tests are not failing, it is the coverage report. But I'm not sure why it's decreasing, it doesn't look like it's decreasing

Sometimes I just don't understand codecov 🤔

**Review by woile — APPROVED** [review]
Thanks for this nice feature!

**Pull request merged**